From 9a779f729e77d4de8ec57aa8153571b5af3ea8ae Mon Sep 17 00:00:00 2001
From: Nizar Haouati <nizar.haouati@intel.com>
Date: Wed, 24 Apr 2013 12:15:40 +0300
Subject: [PORT FROM MAIN] Stk: fix issue in Comprehension Required bit setting

BZ: 2902 13417

The CR (Comprehension Required) bit is set to 1 in the Android code.
It must be set to 0 according to the 3GPP spec.
See ETSI 102.223 chapter 6.8.

This patch sets the CR bit to 0.

Change-Id: I5898dfbc195baca256691f40e03a0123a0422a59
Orig-Change-Id: I4834d796f3da7b9648649cea1e2c82c6d606a37d
Signed-off-by: Nizar Haouati <nizar.haouati@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../android/internal/telephony/cat/CatService.java |    4 ++--
 .../internal/telephony/cat/ResponseData.java       |    6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/java/com/android/internal/telephony/cat/CatService.java b/src/java/com/android/internal/telephony/cat/CatService.java
index e027410..329cefe 100644
--- a/src/java/com/android/internal/telephony/cat/CatService.java
+++ b/src/java/com/android/internal/telephony/cat/CatService.java
@@ -474,14 +474,14 @@ public class CatService extends Handler implements AppInterface {
         buf.write(0x00); // place holder
 
         // device identities
-        tag = 0x80 | ComprehensionTlvTag.DEVICE_IDENTITIES.value();
+        tag = ComprehensionTlvTag.DEVICE_IDENTITIES.value();
         buf.write(tag);
         buf.write(0x02); // length
         buf.write(DEV_ID_KEYPAD); // source device id
         buf.write(DEV_ID_UICC); // destination device id
 
         // item identifier
-        tag = 0x80 | ComprehensionTlvTag.ITEM_ID.value();
+        tag = ComprehensionTlvTag.ITEM_ID.value();
         buf.write(tag);
         buf.write(0x01); // length
         buf.write(menuId); // menu identifier chosen
diff --git a/src/java/com/android/internal/telephony/cat/ResponseData.java b/src/java/com/android/internal/telephony/cat/ResponseData.java
index 2d2e257..4e98502 100644
--- a/src/java/com/android/internal/telephony/cat/ResponseData.java
+++ b/src/java/com/android/internal/telephony/cat/ResponseData.java
@@ -58,7 +58,7 @@ class SelectItemResponseData extends ResponseData {
     @Override
     public void format(ByteArrayOutputStream buf) {
         // Item identifier object
-        int tag = 0x80 | ComprehensionTlvTag.ITEM_ID.value();
+        int tag = ComprehensionTlvTag.ITEM_ID.value();
         buf.write(tag); // tag
         buf.write(1); // length
         buf.write(mId); // identifier of item chosen
@@ -101,7 +101,7 @@ class GetInkeyInputResponseData extends ResponseData {
         }
 
         // Text string object
-        int tag = 0x80 | ComprehensionTlvTag.TEXT_STRING.value();
+        int tag = ComprehensionTlvTag.TEXT_STRING.value();
         buf.write(tag); // tag
 
         byte[] data;
@@ -212,7 +212,7 @@ class DTTZResponseData extends ResponseData {
         }
 
         // DTTZ object
-        int tag = 0x80 | CommandType.PROVIDE_LOCAL_INFORMATION.value();
+        int tag = CommandType.PROVIDE_LOCAL_INFORMATION.value();
         buf.write(tag); // tag
 
         byte[] data = new byte[8];
-- 
1.7.4.1

